arm/Microsoft.Network/applicationGateways/.parameters/parameters.json

```diff
@@ -5,157 +5,356 @@
-            "value": "https://adp-<<namePrefix>>-az-kv-x-001.vault.azure.net/secrets/applicationGatewaySslCertificate/9b670bb436e04eae9acd9b93dca269b6" // ID must be updated for new certificates
+        // "webApplicationFirewallConfiguration": {
+        //     "enabled": true,
+        //     "firewallMode": "Detection",
+        //     "ruleSetType": "OWASP",
+        //     "ruleSetVersion": "3.0",
+        //     "disabledRuleGroups": [],
+        //     "requestBodyCheck": true,
+        //     "maxRequestBodySizeInKb": 128,
+        //     "fileUploadLimitInMb": 100
+        // },
+        // "enableHttp2": true,
+        "backendAddressPools": {
+            "value": [
+                {
+                    "name": "appServiceBackendPool",
+                    "properties": {
+                        "backendAddresses": [
+                            {
+                                "fqdn": "aghapp.azurewebsites.net"
+                            }
+                        ]
+                    }
+                },
+                {
+                    "name": "privateVmBackendPool",
+                    "properties": {
+                        "backendAddresses": [
+                            {
+                                "ipAddress": "10.0.0.4"
+                            }
+                        ]
+                    }
+                }
+            ]
         },
-        "backendPools": {
+        "backendHttpSettingsCollection": {
             "value": [
                 {
-                    "backendPoolName": "appServiceBackendPool",
-                    "backendAddresses": [
-                        {
-                            "fqdn": "aghapp.azurewebsites.net"
-                        }
-                    ]
+                    "name": "appServiceBackendHttpsSetting",
+                    "properties": {
+                        "port": 443,
+                        "protocol": "Https",
+                        "cookieBasedAffinity": "Disabled",
+                        "pickHostNameFromBackendAddress": true,
+                        "requestTimeout": 30
+                    }
                 },
                 {
-                    "backendPoolName": "privateVmBackendPool",
-                    "backendAddresses": [
-                        {
-                            "ipAddress": "10.0.0.4"
+                    "name": "privateVmHttpSetting",
+                    "properties": {
+                        "port": 80,
+                        "protocol": "Http",
+                        "cookieBasedAffinity": "Disabled",
+                        "pickHostNameFromBackendAddress": false,
+                        "requestTimeout": 30,
+                        "probe": {
+                            "id": "/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg/providers/Microsoft.Network/applicationGateways/sxx-az-apgw-x-001/probes/privateVmHttpSettingProbe"
```

Comment:
`sxx` to be replaced with `<<namePrefix>>` in all its occurrences

Reply:
done